e2e/semverutil/semver.go

```diff
@@ -25,7 +25,8 @@ func (fr FeatureReleases) IsSupported(versionStr string) bool {
 	const releasePrefix = "release-"
 	if strings.HasPrefix(versionStr, releasePrefix) {
 		versionStr = versionStr[len(releasePrefix):]
-		versionStr = strings.ReplaceAll(versionStr, "x", "0")
+		// replace x with 999 so the release version is always larger than the others in the release line.
+		versionStr = strings.ReplaceAll(versionStr, "x", "999")
```

Comment:
unrelated fix to ensure semver version comparisons are accurate

Reply:
https://github.com/cosmos/ibc-go/blob/e51b60360be8305f765305e711a29bc4d85f74c3/scripts/update_compatibility_tests.py#L72-L77 prepare for the corner if wrong

Reply:
I think `main` > `release-x.y.z` > `x.y.z`

`main` already gets special treatment in this function